src/Service/Form/Submission/FormSubmissionService.php

```diff
@@ -41,62 +50,132 @@ public function get(string $id): FormSubmission
         return $submission;
     }
 
-    public function createDownload(FormSubmission $formSubmission): string
+    public function createDownload(FormSubmission $formSubmission): StreamedResponse
     {
-        $filesystem = new Filesystem();
-        $tempFile = $filesystem->tempnam(\sys_get_temp_dir(), 'ems_form');
+        $response = new StreamedResponse(function () use ($formSubmission) {
+            $zip = new ZipStream('submission.zip');
+            $data = $formSubmission->getData();
 
-        $zip = new \ZipArchive();
-        $zip->open($tempFile, \ZipArchive::CREATE);
+            $rawJson = \json_encode($data, JSON_UNESCAPED_UNICODE);
+            if (\is_string($rawJson)) {
+                $zip->addFile('data.json', $rawJson);
+            }
 
-        $data = $formSubmission->getData();
-        $data['id'] = $formSubmission->getId();
+            foreach ($formSubmission->getFiles() as $file) {
+                if ($streamRead = $file->getFile()) {
+                    $zip->addFileFromStream($file->getFilename(), $streamRead);
+                } else {
+                    exit('Could not open stream for reading');
+                }
+            }
 
-        $rawJson = \json_encode($data, JSON_UNESCAPED_UNICODE);
-        if (\is_string($rawJson)) {
-            $zip->addFromString('data.json', $rawJson);
-        }
+            $zip->finish();
+        });
 
-        foreach ($formSubmission->getFiles() as $file) {
-            $formFile = $file->getFile();
+        return $response;
+    }
```

Comment:
These function is a duplicate of createDownloadForMultiple

Try this:
``
    public function createDownload(string $formSubmissionId): StreamedResponse
    {
           return $this->createDownloadForMultiple([$SubmissionId]);
    }